cmd/kpod/pull.go

```diff
@@ -0,0 +1,147 @@
+	// cli.BoolFlag{
+	// 	Name:  "all-tags, a",
+	// 	Usage: "Download all tagged images in the repository",
+	// },
+	// cli.BoolFlag{
+	// 	Name:  "disable-content-trust",
+	// 	Usage: "Skip image verification",
+	// },
+	}
+
+	pullDescription = "Pulls an image from a registry and stores it locally.\n" +
+		"An image can be pulled using its tag or digest. If a tag is not\n" +
+		"specified, the image with the 'latest' tag (if it exists) is pulled."
+	pullCommand = cli.Command{
+		Name:        "pull",
+		Usage:       "pull an image from a registry",
+		Description: pullDescription,
+		Flags:       pullFlags,
+		Action:      pullCmd,
+		ArgsUsage:   "",
+	}
+)
+
+// pullCmd gets the data from the command line and calls pullImage
+// to copy an image from a registry to a local machine
+func pullCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) == 0 {
+		logrus.Errorf("an image name must be specified")
+		return nil
+	}
+	if len(args) > 1 {
+		logrus.Errorf("too many arguments. Requires exactly 1")
+		return nil
+	}
+	image := args[0]
+
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	allTags := false
+	// if c.IsSet("all-tags") {
+	// 	allTags = c.Bool("all-tags")
+	// }
+	disableContentTrust := true
+	// if c.IsSet("disable-content-trust") {
+	// 	disableContentTrust = c.Bool("disable-content-trust")
+	// }
+
+	options := PullOptions{
+		FromImage:           image,
+		AllTags:             allTags,
+		DisableContentTrust: disableContentTrust,
+	}
+
+	systemContext := getSystemContext("")
+
+	err = pullImage(store, options, systemContext)
+	if err != nil {
+		logrus.Errorf("error pulling image from %q: %v", options.FromImage, err)
+	}
+	return nil
+}
+
+// pullImage copies the image from the source to the destination
+func pullImage(store storage.Store, options PullOptions, sc *types.SystemContext) error {
+	imgName := options.FromImage
+	defaultName := DefaultRegistry + imgName
+	var fromName string
+	var tag string
+
+	srcRef, err := alltransports.ParseImageName(defaultName)
+	if err != nil {
+		srcRef2, err2 := alltransports.ParseImageName(imgName)
+		if err2 != nil {
+			return errors.Wrapf(err2, "error parsing image name %q", imgName)
+		}
+		srcRef = srcRef2
+	}
+
+	ref := srcRef.DockerReference()
+	if ref != nil {
+		imgName = srcRef.DockerReference().Name()
+		fromName = imgName
+		tagged, ok := srcRef.DockerReference().(reference.NamedTagged)
+		if ok {
+			imgName = imgName + ":" + tagged.Tag()
+			tag = tagged.Tag()
+		}
+	}
+
+	destRef, err := is.Transport.ParseStoreReference(store, imgName)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing full image name %q", imgName)
+	}
+
+	policy, err := signature.DefaultPolicy(sc)
+	if err != nil {
+		return err
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(tag + ": pulling from " + fromName)
+
+	err = cp.Image(policyContext, destRef, srcRef, getCopyOptions(os.Stderr))
```

Comment:
why logging to stderr?

You can directly do:
```
return  cp.Image(policyContext, destRef, srcRef, getCopyOptions(os.Stderr))
```
w/o the `err = ` assignment. But first, I think this should be stdout (@mrunalp?)